x-pack/plugins/cloud_security_posture/server/create_indices/create_indices.ts

```diff
@@ -4,72 +4,153 @@
+  LATEST_FINDINGS_INDEX_PATTERN,
+  LATEST_FINDINGS_INDEX_TEMPLATE_NAME,
 } from '../../common/constants';
 import { createPipelineIfNotExists } from './create_processor';
+import { benchmarkScoreMapping } from './benchmark_score_mapping';
 
 // TODO: Add integration tests
+
 export const initializeCspIndices = async (esClient: ElasticsearchClient, logger: Logger) => {
   await createPipelineIfNotExists(esClient, CSP_INGEST_TIMESTAMP_PIPELINE, logger);
+
   return Promise.all([
-    createIndexIfNotExists(
-      esClient,
-      LATEST_FINDINGS_INDEX_NAME,
-      LATEST_FINDINGS_INDEX_DEFAULT_NS,
-      latestFindingsMapping,
-      {},
-      logger
-    ),
-    createIndexIfNotExists(
-      esClient,
-      BENCHMARK_SCORE_INDEX_NAME,
-      BENCHMARK_SCORE_INDEX_DEFAULT_NS,
-      benchmarkScoreMapping,
-      { default_pipeline: CSP_INGEST_TIMESTAMP_PIPELINE },
-      logger
-    ),
+    createLatestFindingsIndex(esClient, logger),
+    createBenchmarkScoreIndex(esClient, logger),
   ]);
 };
 
-export const createIndexIfNotExists = async (
+const createBenchmarkScoreIndex = async (esClient: ElasticsearchClient, logger: Logger) => {
+  try {
+    const INDEX_TEMPLATE_V830 = 'cloud_security_posture.scores';
+    await deleteIndexTemplateSafe(esClient, logger, INDEX_TEMPLATE_V830);
+
+    // We always want to keep the index template updated
+    await esClient.indices.putIndexTemplate({
+      name: BENCHMARK_SCORE_INDEX_TEMPLATE_NAME,
+      index_patterns: BENCHMARK_SCORE_INDEX_PATTERN,
+      template: {
+        mappings: benchmarkScoreMapping,
+        settings: {
+          default_pipeline: CSP_INGEST_TIMESTAMP_PIPELINE,
+          lifecycle: {
+            // This is the default lifecycle name, it is named on the data-stream type (e.g, logs/ metrics)
+            name: 'logs',
+          },
+        },
+      },
+      _meta: {
+        package: {
+          name: CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+        },
+        managed_by: 'cloud_security_posture',
+        managed: true,
+      },
+      priority: 500,
+    });
+
+    await createIndexSafe(esClient, logger, BENCHMARK_SCORE_INDEX_DEFAULT_NS);
+  } catch (e) {
+    logger.error(
+      `Failed to upsert index template [Template: ${BENCHMARK_SCORE_INDEX_TEMPLATE_NAME}]`
+    );
+    logger.error(e);
+  }
+};
+
+const createLatestFindingsIndex = async (esClient: ElasticsearchClient, logger: Logger) => {
+  try {
+    const INDEX_TEMPLATE_V830 = 'cloud_security_posture.findings_latest';
+    await deleteIndexTemplateSafe(esClient, logger, INDEX_TEMPLATE_V830);
+
+    // We want that our latest findings index template would be identical to the findings index template
+    const findingsIndexTemplateResponse = await esClient.indices.getIndexTemplate({
+      name: FINDINGS_INDEX_NAME,
+    });
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { template, composed_of, _meta } =
+      findingsIndexTemplateResponse.index_templates[0].index_template;
```

Comment:
we might need to check for `index_templates` array has an object on member `0` before destructuring ?

Reply:
There's a try-catch on this one, so for now i'm ok with it
Thats because we do expect that the index template would be there and if not - it is an error